documentation/dotnet-monitor.md

```diff
@@ -0,0 +1,309 @@
+
+To collect a trace of the target process, run the following command:
+
+In PowerShell,
+
+```powershell
+PS> Start-Process http://localhost:52323/trace
+```
+
+In bash,
+```bash
+$ wget --content-disposition http://localhost:52323/trace
+```
+
+The resulting `.nettrace` file can be analyzed with both Visual Studio and PerfView.
+
+### Logs
+
+The `/logs` endpoint will stream logs from the target process for a duration of 30 seconds.
+
+The duration of collection can be customized via the `durationSeconds` querystring parameter. The logs endpoint is capable of returning either newline delimited JSON([application/x-ndjson](https://github.com/ndjson/ndjson-spec)) or the Event stream format([text/event-stream](https://developer.mozilla.org/docs/Web/API/Server-sent_events/Using_server-sent_events#Event_stream_format)) based on the specified `Accept` header in the HTTP request.
+
+To start streaming log from the target process, run the following command:
+
+In PowerShell,
+
+```powershell
+PS> Start-Process http://localhost:52323/logs
+```
+
+In bash,
+```bash
+$ curl -H "Accept:application/x-ndjson" http://localhost:52323/logs --no-buffer
+```
+
+### Metrics
+
+The `/metrics` endpoint will return a snapshot of runtime and ASP.NET Core metrics in the [prometheus exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format). Unlike the other diagnostics endpoints, the metrics endpoint will not be available if `dotnet-trace` detects more than one target process. In addition to being accessible via the URLs configured via the `--urls` parameters, the metrics endpoint is also accessible from the URLs configured via the `--metricUrls`. When running in Kubernetes, it may be suitable to expose the metrics URL to other services in your cluster to allow them to scrape metrics.
+
+While metrics collection is enabled by default when `dotnet-monitor` detects exactly one target process, it can configured to disable to collection of metrics entirely via the `--metrics` parameter. In the example below, metrics collection will not be enabled.
+
+````bash
+dotnet monitor collect --metrics false
+````
+
+When deploying in-cluster, a common patter to collect metrics is to use Prometheus or another monitoring tool to scrape the metrics endpoint exposed by your application. As an example, when running in Azure Kubernetes Services(AKS), you can [configure Azure Monitor to scrape prometheus metrics](https://docs.microsoft.com/azure/azure-monitor/insights/container-insights-prometheus-integration) exposed by `dotnet-monitor`. By following the instructions in the linked document, you can enable Azure Monitor to [enable monitoring pods](https://gist.github.com/shirhatti/0222017e8e2fdb481f735002f7bd72f7/revisions) that have been [annotated](https://gist.github.com/shirhatti/ad7a986137d7ca6b1dc094a3e0a61a0d#file-hello-world-deployment-yaml-L18-L19).
```

Comment:
Metrics has some additional configurations (that can be configured with environment variables or config maps):
DotnetMonitor_PrometheusConfiguration__MetricCount - maximum metric count to store in memory. defaults to 3.
DotnetMonitor_PrometheusConfiguration__UpdateIntervalSeconds - how frequently we poll metrics. defaults to 10 seconds.

Reply:
Maybe omit it for brevity? We should definitely have it in the docs, but I guess it is not needed in the blog post

Reply:
Sounds good.